pkg/arvo/app/permission-group-hook.hoon

```diff
@@ -0,0 +1,179 @@
+::
+++  poke-permission-group-hook-action
+  |=  act=permission-group-hook-action
+  ^-  (quip move _this)
+  ?.  =(src.bol our.bol)
+    [~ this]
+  ?-  -.act
+      %associate
+    (handle-associate group.act permissions.act)
+  ::
+      %dissociate
+    (handle-dissociate group.act permissions.act)
+  ::
+  ==
+::
+++  handle-associate
+  |=  [group=path permissions=(set path)]
+  ^-  (quip move _this)
+  ::  TODO: remove the requirement for permission paths to exist
+  =/  perm-list  ~(tap in permissions)
+  =/  perm-length  (lent perm-list)
+  =/  checked-paths
+    %+  skim  perm-list
+    |=  pax=path
+    ^-  ?
+    ?~  (permission-scry pax)
+      %.n
+    %.y
+  ?.  =((lent checked-paths) perm-length)
+    [~ this]
+  =/  perms  (~(get by relation) group)
+  ::  if relation does not exist, create it and subscribe.
+  ?~  perms
+    =/  group-path  [%group group]
+    :_  this(relation (~(put by relation) group permissions))
+    [ost.bol %peer group-path [our.bol %group-store] group-path]~
+  =.  u.perms  (~(uni in u.perms) permissions)
+  :-  ~
+  this(relation (~(put by relation) group u.perms))
+::
+++  handle-dissociate
+  |=  [group=path permissions=(set path)]
+  ^-  (quip move _this)
+  =/  perms  (~(get by relation) group)
+  ?~  perms
+    [~ this]
+  =.  permissions  (~(del in u.perms) permissions)
+  ?~  permissions
+    :_  this(relation (~(del by relation) group))
+    :~  (group-pull [%group group])
+    ==
+  :-  ~
+  this(relation (~(put by relation) group permissions))
+::
+++  diff-group-update
+  |=  [wir=wire diff=group-update]
+  ^-  (quip move _this)
+  ?-  -.diff
+      %keys
+    [~ this]
+      %bundle
+    [~ this]
+      %path
+    ::  set all permissions paths
+    =/  perms  (~(get by relation) pax.diff)
+    ?~  perms
+      [~ this]
+    :_  this
+    %+  turn  ~(tap in u.perms)
+    |=  =path
+    ^-  move
+    (permission-poke path [%add path members.diff])
+  ::
+      %add
+    ::  set all permissions paths
+    =/  perms  (~(get by relation) pax.diff)
+    ?~  perms
+      [~ this]
+    :_  this
+    %+  turn  ~(tap in u.perms)
+    |=  =path
+    ^-  move
+    (permission-poke path [%add path members.diff])
+  ::
+      %remove
+    ::  set all permissions paths
+    =/  perms  (~(get by relation) pax.diff)
+    ?~  perms
+      [~ this]
+    :_  this
+    %+  turn  ~(tap in u.perms)
+    |=  =path
+    ^-  move
+    (permission-poke path [%remove path members.diff])
+  ::
+      %unbundle
+    ::  pull subscriptions
+    =/  perms  (~(get by relation) pax.diff)
+    ?~  perms
+      :_  this(relation (~(del by relation) pax.diff))
+      :~  (group-pull [%group pax.diff])
+      ==
+    :_  this(relation (~(del by relation) pax.diff))
+    :-  (group-pull [%group pax.diff])
+    %+  turn  ~(tap in u.perms)
+    |=  =path
+    ^-  move
+    (permission-poke path [%delete path])
+  ::
+  ==
+::
+++  quit
+  |=  wir=wire
+  ^-  (quip move _this)
+  ::  TODO: implement resubscribe handler
```

Comment:
With this kind of rare error that you're explicitly not handling yet, I recommend a one-line printf so if people report bad behavior you can guess why.

Reply:
This is no longer relevant since I'm correctly handling the ++reap case. It correctly no-ops.